examples/differentially_private_sgd.py

```diff
@@ -0,0 +1,270 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""JAX trains a differentially private conv net on MNIST 30X faster than TF.
+
+This script contains a JAX implementation of Differentially Private Stochastic
+Gradient Descent (https://arxiv.org/abs/1607.00133). DPSGD requires clipping
+the per-example parameter gradients, which is non-trivial to implement
+efficiently for convolutional neural networks. For example, the reference
+tensorflow implementation
+(https://github.com/tensorflow/privacy/tree/master/tutorials) takes upwards of 2
```

Comment:
Same here; maybe just rephrase to "See also the reference TensorFlow implementation: <url>." without timing comparison.

Reply:
948def8 fixes this